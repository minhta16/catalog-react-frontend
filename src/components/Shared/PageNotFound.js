import React from 'react';
import { Link } from 'react-router-dom';
import { Typography, Button, Grid, Paper, Container } from '@material-ui/core';

const PageNotFound = () => (
  <Container maxWidth="lg">
    <Paper style={{ padding: '1rem' }}>
      <Grid container spacing={2} justify="center">
        <Grid item xs={12}>
          <Typography variant="h3">Page Not Found</Typography>
        </Grid>
        <Grid item xs={12}>
          <Typography variant="body1">
            The link you followed may be broken, or the page may have been removed.
          </Typography>
        </Grid>
        <Grid item xs={12}>
          <Button variant="contained" color="primary" component={Link} exact="true" to="/">
            Go back to Home
          </Button>
        </Grid>
      </Grid>
    </Paper>
  </Container>
);

export default PageNotFound;
