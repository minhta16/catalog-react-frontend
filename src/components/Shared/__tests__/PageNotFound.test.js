/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import PageNotFound from 'components/Shared/PageNotFound';

describe('components/LoginDialog', () => {
  let wrapper;
  // let warning;

  const setup = () => {
    wrapper = shallow(<PageNotFound />);
  };

  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });
});
