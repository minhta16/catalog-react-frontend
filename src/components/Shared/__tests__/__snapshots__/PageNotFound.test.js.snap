// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LoginDialog should render correctly 1`] = `
<WithStyles(ForwardRef(Container))
  maxWidth="lg"
>
  <WithStyles(ForwardRef(Paper))
    style={
      Object {
        "padding": "1rem",
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="center"
      spacing={2}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <WithStyles(ForwardRef(Typography))
          variant="h3"
        >
          Page Not Found
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <WithStyles(ForwardRef(Typography))
          variant="body1"
        >
          The link you followed may be broken, or the page may have been removed.
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          color="primary"
          component={[Function]}
          exact="true"
          to="/"
          variant="contained"
        >
          Go back to Home
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Paper))>
</WithStyles(ForwardRef(Container))>
`;
